def create_VPN_Users(self, rand_name=True, api_client=None):
    'Creates VPN users for the network'
    self.debug(('Creating VPN users for account: %s' % self.account.name))
    if (api_client is None):
        api_client = self.apiclient
    try:
        vpnuser = VpnUser.create(api_client, self.services['vpn_user']['username'], self.services['vpn_user']['password'], account=self.account.name, domainid=self.account.domainid, rand_name=rand_name)
        self.debug('Verifying the remote VPN access')
        vpn_users = VpnUser.list(self.apiclient, id=vpnuser.id, listall=True)
        self.assertEqual(isinstance(vpn_users, list), True, 'List VPNs shall return a valid response')
        return vpnuser
    except Exception as e:
        self.fail(('Failed to create remote VPN users: %s' % e))
